### PR TITLE
Rename blockdims/blockshape to chunks

### PR DESCRIPTION
**Pull request by mrocklin**
Old

```python
x = da.from_array(x, blockshape=(4, 5))
or
x = da.from_array(x, blockdims=((4, 4, 4), (5, 5, 5, 3)))
```

New

```python
x = da.from_array(x, chunks=(4, 5))
or
x = da.from_array(x, chunks=((4, 4, 4), (5, 5, 5, 3))
```

I also remove into and blaze implementations.  These have been moved or are being moved to odo/blaze repositories.

### CODE REVIEW

File: dask/array/core.py [hunk 1]
```diff
@@ -536,7 +535,9 @@ def blockdims_from_blockshape(shape, blockshape):
     ((4, 4, 2), (3, 3, 3, 1))
     """
     if blockshape is None:
```
Comment (shoyer) [inline dask/array/core.py]: looks like this function and its arguments needs a rename. perhaps `expand_chunks`?

File: dask/array/core.py [hunk 2]
```diff
@@ -536,7 +535,9 @@ def blockdims_from_blockshape(shape, blockshape):
...
-        raise ValueError("Must supply a blockshape= keyword argument")
+        raise ValueError("Must supply chunks= keyword argument")
```
Comment (shoyer) [inline dask/array/core.py]: `TypeError` might be more appropriate?

File: dask/array/core.py (hunk 1 above)
Comment (shoyer) [inline dask/array/core.py]: Actually, perhaps you can get ride of it entirely? I see you added `normalize_chunks`.

File: dask/array/core.py [hunk 3]
```diff
@@ -1454,11 +1482,11 @@ def insert(arr, obj, values, axis):
...
 
-    values_blockdims = tuple(values_bd if axis == n else arr_bd
-                             for n, (arr_bd, values_bd)
-                             in enumerate(zip(arr.blockdims,
-                                              values.blockdims)))
-    values = values.reblock(values_blockdims)
+    values_chunks = tuple(values_bd if axis == n else arr_bd
+                          for n, (arr_bd, values_bd)
+                          in enumerate(zip(arr.chunks,
+                                              values.chunks)))
```
Comment (shoyer) [inline dask/array/core.py]: nit: indentation

Comment (mrocklin) [inline dask/array/core.py]: Fixed.

File: dask/array/core.py (hunk 1 above)
Comment (mrocklin) [inline dask/array/core.py]: `normalize_chunks` uses this function.  I do still like these names when we need to distinguish between the two concepts.

File: dask/array/core.py (hunk 2 above)
Comment (mrocklin) [inline dask/array/core.py]: Changed to `TypeError`

File: dask/array/core.py (hunk 1 above)
Comment (shoyer) [inline dask/array/core.py]: agreed!

**Comment by shoyer** [comment]
will renaming reblock -> rechunk be another PR?

**Comment by mrocklin** [comment]
> will renaming reblock -> rechunk be another PR?

Done

**Comment by mrocklin** [comment]
Also, now that the Blaze/into code has been moved out of dask the whole test suite is quite snappy again 330 tests in 6 seconds.  Almost back to real time speeds.

**Comment by shoyer** [comment]
The docs also need an update, at the very least this page: http://dask.readthedocs.org/en/latest/array-creation.html

**Comment by shoyer** [comment]
Probably also worth describing the auto-expanding `chunks` attribute.

**Comment by mrocklin** [comment]
I've descrbed `chunks` behavior in the docs and in `from_array`.

**Comment by mrocklin** [comment]
Any further feedback @shoyer

**Comment by mrocklin** [comment]
Planning to merge soon if no comment.

**Comment by shoyer** [comment]
Let me give it a quick read over again....

**Comment by shoyer** [comment]
Looks good to me! We may have still missed something, but that will be easier to figure out in a followup.

**Comment by shoyer** [comment]
Thanks!

**Comment by cowlicks** [comment]
:+1:

**Pull request merged**